sequence_processing_pipeline/NuQCJob.py

```diff
@@ -0,0 +1,368 @@
+        self.project_data = metadata['projects']
+        self.needs_trimming = metadata['needs_adapter_trimming']
+        self.nprocs = 16 if nprocs > 16 else nprocs
+        self.chemistry = metadata['chemistry']
+        # instead of a list of individual .mmi files,  minimap_database_paths
+        # is now a path to a directory of .mmi files.
+        self.minimap_database_paths = minimap_database_paths
+        self.queue_name = queue_name
+        self.node_count = node_count
+        self.wall_time_limit = wall_time_limit
+        self.jmem = jmem
+        self.fastp_path = fastp_path
+        self.minimap2_path = minimap2_path
+        self.samtools_path = samtools_path
+        self.qiita_job_id = qiita_job_id
+        self.pool_size = pool_size
+        self.suffix = 'fastq.gz'
+
+        ### BEGIN NEW PARAMS
+        # sizes in GB
+        self.max_file_list_size_in_gb = 2  # max size of GB data to process in a job
+        self.data_location = "/panfs/dtmcdonald/human-depletion/t2t-only/*/*"
+
+        datestamp = date.today().strftime("%Y.%m.%d")
+
+        self.batch_prefix = join(self.output_path,
+                                 f"1-hd-split-pangenome-pe-{datestamp}")
+
+        self.output = "/panfs/dtmcdonald/human-depletion/pangenome-adapter-filter"
+        self.tmpdir = "/scratch/tcga_compute/tmp"
+        ### END NEW PARAMS
+
+        self.minimum_bytes = 3100
+
+        if not isinstance(self.needs_trimming, bool):
+            raise ValueError("needs_adapter_trimming must be boolean.")
+
+        # Validate project settings in [Bioinformatics]
+        for project in self.project_data:
+            if project['ForwardAdapter'] == 'NA':
+                project['ForwardAdapter'] = None
+
+            if project['ReverseAdapter'] == 'NA':
+                project['ReverseAdapter'] = None
+
+            if project['ForwardAdapter'] is None:
+                if project['ReverseAdapter'] is not None:
+                    raise ValueError(("ForwardAdapter is declared but not "
+                                      "ReverseAdapter."))
+
+            if project['ReverseAdapter'] is None:
+                if project['ForwardAdapter'] is not None:
+                    raise ValueError(("ReverseAdapter is declared but not "
+                                      "ForwardAdapter."))
+
+            if not isinstance(project['HumanFiltering'], bool):
+                raise ValueError("needs_adapter_trimming must be boolean.")
+
+    def _filter_empty_fastq_files(self, filtered_directory,
+                                  empty_files_directory,
+                                  minimum_bytes):
+        '''
+        Filters out and moves fastq files that are below threshold.
+        :param filtered_directory:
+        :param empty_files_directory:
+        :param minimum_bytes:
+        :return:
+        '''
+        empty_list = []
+
+        for entry in listdir(filtered_directory):
+            if '_R1_' in entry:
```

Comment:
this is not safe as there are also instances of ".R1."

Reply:
modified by Daniel.